app/src/main/java/fragments/TempFeedFragment.java

```diff
@@ -168,7 +195,49 @@ public void onItemSelected(AdapterView<?> parent, View view, int pos, long id) {
     @Override
     public void onNothingSelected(AdapterView<?> parent) {
         current_beach = (BeachGroup) parent.getItemAtPosition(0);
+        getWeatherDetails(current_beach.getKeyLocationid());
         QueryUtils.queryShortPosts(allPosts, adapter, current_beach);
         bindDescription();
     }
+
+    public void getWeatherDetails(String locationID){
+        String temp_url = url + "?id=" + locationID + "&appid=" + api_key;
+        StringRequest stringRequest = new StringRequest(Request.Method.POST, temp_url,
+                new Response.Listener<String>() {
+                    @Override
+                    public void onResponse(String response) {
+                        try {
+                            JSONObject jsonResponse = new JSONObject(response);
+                            Log.i(TAG, "response: " + response);
+
+                            JSONArray jsonWeatherArray = jsonResponse.getJSONArray("weather");
+
+                            JSONObject jsonWeatherObject = jsonWeatherArray.getJSONObject(0);
+                            JSONObject jsonMainObject = jsonResponse.getJSONObject("main");
+                            JSONObject jsonSysObject = jsonResponse.getJSONObject("sys");
+
+                            description = jsonWeatherObject.getString("description");
+                            double tempKelvin = jsonMainObject.getDouble("temp");
+                            temperature = TempUtils.kelvin_to_fahrenheit(Double.toString(tempKelvin));
+                            long sunsetUTC = jsonSysObject.getLong("sunset");
+                            long locationTimeZone = jsonResponse.getLong("timezone");
+                            // need to account for location timezone and my timezone
+                            sunsetTime = TimeUtils.unixToUTC(sunsetUTC + locationTimeZone - myTimeZone);
+
+                            tvWeather.setText(description);
+                            tvAirTemp.setText(temperature);
+                            tvSunsetTime.setText(sunsetTime);
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                Log.i(TAG, "Error loading in weather data.");
+            }
+        });
+        RequestQueue requestQueue = Volley.newRequestQueue(getContext());
```

Comment:
nice. volley does a good job handling your threading. while you are initiating your request on the UI thread, volley will actually process this work on a background thread.